cmd/cql/main.go

```diff
@@ -240,24 +258,65 @@ func init() {
 	flag.BoolVar(&waitTxConfirmation, "wait-tx-confirm", false, "Wait for transaction confirmation")
 }
 
+func serverAsShardChainExplorer(webAddr string) {
+	service, httpServer, err := observer.StartObserver(webAddr, version)
+	if err != nil {
+		log.WithError(err).Fatal("start observer failed")
+	}
+	<-stopCh
+
+	_ = observer.StopObserver(service, httpServer)
+
+	log.Info("observer stopped")
+}
+
 func main() {
+	var err error
+	// set random
+	rand.Seed(time.Now().UnixNano())
+
 	flag.Parse()
+	log.SetStringLevel(logLevel, log.InfoLevel)
+	if tmpPath == "" {
+		tmpPath = os.TempDir()
+	}
+	logPath := filepath.Join(tmpPath, "covenant_explorer.log")
+	bgLog, err := os.OpenFile(logPath, os.O_CREATE|os.O_WRONLY|os.O_TRUNC, 0644)
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "open log file failed: %s, %v", logPath, err)
+		os.Exit(-1)
+	}
+	log.SetOutput(bgLog)
+	log.SetStringLevel(bgLogLevel, log.InfoLevel)
+
+	cLog = logrus.New()
+
 	if showVersion {
 		fmt.Printf("%v %v %v %v %v\n",
 			name, version, runtime.GOOS, runtime.GOARCH, runtime.Version())
 		os.Exit(0)
 	}
-	log.Infof("cql build: %#v\n", version)
+	cLog.Infof("cql build: %#v\n", version)
 
 	configFile = utils.HomeDirExpand(configFile)
 
-	var err error
+	if explorerAddr != "" {
+		var err error
+		conf.GConf, err = conf.LoadConfig(configFile)
+		if err != nil {
+			log.WithField("config", configFile).WithError(err).Fatal("load config failed")
+		}
 
-	// init covenantsql driver
-	if err = client.Init(configFile, []byte(password)); err != nil {
-		log.WithError(err).Error("init covenantsql client failed")
-		os.Exit(-1)
+		serverAsShardChainExplorer(explorerAddr)
+		defer close(stopCh)
 		return
+	} else {
```

Comment:
**[golint]** <sub>reported by [reviewdog](https://github.com/haya14busa/reviewdog) :dog:</sub>
if block ends with a return statement, so drop this else and outdent its block